pilot/pkg/model/xds_cache.go

```diff
@@ -277,40 +257,62 @@ func (l *lruCache) Add(entry XdsCacheEntry, pushReq *PushRequest, value *discove
 	}
 
 	dependentConfigs := entry.DependentConfigs()
-	dependentTypes := entry.DependentTypes()
-	toWrite := cacheValue{value: value, token: token, dependentConfigs: dependentConfigs, dependentTypes: dependentTypes}
+	toWrite := cacheValue{value: value, token: token, dependentConfigs: dependentConfigs}
 	l.store.Add(k, toWrite)
 	l.token = token
 	l.updateConfigIndex(k, dependentConfigs)
-	l.updateTypesIndex(k, dependentTypes)
 	size(l.store.Len())
 }
 
 type cacheValue struct {
 	value            *discovery.Resource
 	token            CacheToken
 	dependentConfigs []ConfigHash
-	dependentTypes   []kind.Kind
 }
 
-func (l *lruCache) Get(entry XdsCacheEntry) (*discovery.Resource, bool) {
+func (l *lruCache) Get(entry XdsCacheEntry) *discovery.Resource {
 	if !entry.Cacheable() {
-		return nil, false
+		return nil
 	}
 	k := entry.Key()
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	cv, ok := l.store.Get(k)
 	if !ok {
 		miss()
-		return nil, false
+		return nil
 	}
 	if cv.value == nil {
 		miss()
-		return nil, false
+		return nil
 	}
 	hit()
-	return cv.value, true
+	return cv.value
+}
+
+// GetWithRequest return the cached value generated by newer push request than the given one.
+func (l *lruCache) GetWithRequest(entry XdsCacheEntry, pushReq *PushRequest) *discovery.Resource {
```

Comment:
Can you add an internal function that these two can use without repeating the code?

Reply:
yes